lesson_02_homework.rb

```diff
@@ -1,46 +1,80 @@
 # Code Reading
 
+# declaring full_name method with three arguments
 def full_name(first_name, last_name, title)
-  # Example comment line
-
+  
+  # assigning variable "name" equal to nil
   name = nil
-
+  # declaring if/else statement, if title, first_name and last_name arguments are called, ruby runs the equation below
   if title && first_name && last_name
+    # equaling name to (title + " " + first_name + " " + last_name)
     name = title + " " + first_name + " " + last_name
+  # if title and last_name arguments are only called, ruby runs the equation below
   elsif title && last_name
+    # equaling name to (title + " " + last_name)
     name = title + " " + last_name
+  # if first_name and last_name arguments are only called, ruby runs the equation below
   elsif first_name && last_name
+    # equaling name to (first_name + " " + last_name)
     name = first_name + " " + last_name
+  # if first_name argument is only called, ruby runs the equation below 
   elsif first_name
+    # equaling name to (first_name)
     name = first_name
+  # if all else fails, ruby executes raise
   else
+    # ruby runs raise error message
     raise "Oh no, that doesn't look like a name"
+  # ends if/else statement
   end
-
+  # returns name
   return name
+# ends full_name method
 end
 
+puts full_name('Stephen', 'Bienkiewicz', 'Mr.')
+
+
 # Fix Broken Code
 
 def full_name(f, l)
```

Comment:
🔨 @stephenbienkiewicz  the parameter names should be more explicit `f` and `l` do not describe what type of information the variable is expected to hold